internal/static/parse.go

```diff
@@ -0,0 +1,72 @@
+package static
+
+import (
+	"go/constant"
+	"go/types"
+
+	"github.com/dogmatiq/configkit"
+	"github.com/dogmatiq/configkit/internal/entity"
+	"github.com/dogmatiq/configkit/internal/typename/gotypes"
+	"golang.org/x/tools/go/ssa"
+)
+
+// parse parses the member of the package and populates it into
+// entity.Application.
+//
+// The second parameter is the type that actually implements
+// github.com/dogmatiq/dogma.Application interface.
+func parse(m ssa.Member, typ types.Type) *entity.Application {
+	app := &entity.Application{
+		TypeNameValue: gotypes.NameOf(typ),
+	}
+
+	fn := m.Package().Prog.LookupMethod(
+		typ,
+		m.Package().Pkg,
+		"Configure",
+	)
+
+	// Since `.Configure()` is the method, the first parameter is the receiver,
+	// so we need to extract the second paramenter as
+	// `dogma.ApplicationConfigurer`
+	configurer := fn.Params[1]
+
+	for _, c := range filterCalls(fn) {
+		// Check if we are calling the methods on the configurer.
+		if c.Common().Value.Name() == configurer.Name() {
+			switch c.Common().Method.Name() {
+			case "Identity":
+				app.IdentityValue = parseIdentity(c)
+			}
+		}
+	}
+
+	return app
+}
+
+// filterCalls filters and returns all function/method call instructions within
+// the given function.
+func filterCalls(fn *ssa.Function) (calls []*ssa.Call) {
+	for _, b := range fn.Blocks {
+		for _, i := range b.Instrs {
+			if c, ok := i.(*ssa.Call); ok {
+				calls = append(calls, c)
+			}
+		}
+	}
+
+	return
```

Comment:
As a matter of consistent style please avoid naked returns. I know they're not a "big deal" on small functions but rather than trying define what "small" is let's just avoid them altogether.

Likewise, only used named return types if you need to refer to them in documentation or they need to be assigned during a `defer`.

Reply:
This issue is also addressed in https://github.com/dogmatiq/configkit/pull/118/commits/b4777f379b5ec0bd39486dad955fee964c8428c1.